sdk/python/tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -1,24 +1,25 @@
 import tempfile
 import uuid
 from contextlib import contextmanager
+from dataclasses import replace
+from datetime import datetime, timedelta
 from pathlib import Path
-from typing import Dict, List, Union
+from typing import Dict, List, Optional, Tuple, Union
 
+import pandas as pd
 import pytest
 from attr import dataclass
 
-from feast import FeatureStore, RepoConfig, importer
+from feast import FeatureStore, FeatureView, RepoConfig, driver_test_data, importer
+from feast.data_source import DataSource
 from tests.data.data_creator import create_dataset
 from tests.integration.feature_repos.universal.data_source_creator import (
     DataSourceCreator,
 )
-from tests.integration.feature_repos.universal.entities import driver
-from tests.integration.feature_repos.universal.feature_views import (
-    correctness_feature_view,
-)
+from tests.integration.feature_repos.universal.entities import customer, driver
 
 
-@dataclass
+@dataclass(frozen=True, str=True)
```

Comment:
what is str=True doing here? dont see it in documentation around dataclass?

Reply:
This generates a human readable string for pytest summaries:

```
===================================================================================== short test summary info =====================================================================================
FAILED sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py::test_historical_features_from_bigquery_sources[TestRepoConfig(provider='local', online_store='sqlite', offline_store_creator='tests.integration.feature_repos.universal.data_sources.file.FileDataSourceCreator', full_feature_names=True, infer_event_timestamp_col=True)]
```

Reply:
It changed to `repr` cause I was using dataclasses from `attr` by mistake. Fixed now.